### PR TITLE
Android 3.0.0 migration update

### PR DESCRIPTION
**Pull request by cacheung**
Android 3.0.0 migration update

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/.gitignore
Comment (kevinlind) [inline code/.gitignore:1]: Update the root `.gitignore` instead, then you can simply remove this file.

File: code/app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+}
+
+android {
+    namespace = "com.adobe.marketing.edge.identity.testapp"
+
+    compileSdk = BuildConstants.Versions.COMPILE_SDK_VERSION
+
+    defaultConfig {
+        applicationId = "com.adobe.marketing.edge.identity.tester"
```
Comment (kevinlind) [inline code/app/build.gradle.kts:25]: For simplicity, should we set the `applicationId` the same as `namespace`?

File: code/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <application
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
-        android:name="com.adobe.marketing.edge.identity.app.EdgeIdentityApplication"
+        android:name="com.adobe.marketing.edge.identity.testapp.EdgeIdentityApplication"
```
Comment (kevinlind) [inline code/app/src/main/AndroidManifest.xml:8]: Change to just `.EdgeIdentityApplication` (note the "."). If you `namespace` is correctly set in the build.gradle.kts then just specifying the class name should resolve and build.

File: .gitignore [hunk 3]
```diff
@@ -6,10 +6,25 @@ bin/
 build/
 ci/
 
+# Gradle files
+.gradle/
+build/
+ci/
+jacoco.exec
```
Comment (kevinlind) [inline .gitignore:13]: `jacoco.exec` is already included below at line 38.

File: .gitignore [hunk 4]
```diff
@@ -6,10 +6,25 @@ bin/
...
 # IntelliJ
 *.iml
 **/.idea/
 
+# Log Files
+*.log
+
+# .Idea file
+ code/.idea*
+
```
Comment (kevinlind) [inline .gitignore:24]: IntelliJ files are already included in lines 15-17.

File: .gitignore [hunk 5]
```diff
@@ -6,10 +6,25 @@ bin/
 build/
 ci/
 
+# Gradle files
+.gradle/
+build/
+ci/
```
Comment (kevinlind) [inline .gitignore:12]: These files are already included above on lines 5-7. I also don't think we should ignore `.gradle` files.

Comment (cacheung) [reply .gitignore]: Thanks for catching these @kevinlind .  Updated.

**Review by kevinlind — APPROVED** [review]

**Pull request merged**